# **Assignment 2 Sanaz(Math 537, Fall 2022)**

# Sanaz Hami Hassan Kiyadeh
# shkiyadeh@crimson.ua.edu

# Problem 2





## Part c

Write a Python script to use your expression from Part~(b) to evaluate $f(0.001)$ to $8$ significant digits.




In [8]:
# Evaluate using while loop.
import math
# Initialize i.
i = 1
# Define x.
x = 0.001
f_1 = 0
# Print header for the output table.
print("%+5s | %-10s | %-15s" %("i", "While loop", "Explicit Formula"))
print("=========================================================")

while (i <= 8):
    f_1 += ((2**i)/math.factorial(i))*x**(i-1)
    # increment i.
    i += 1
# Evaluate f explicitely.
x = 0.001
a = math.e**(2*x) - 1
b = x
f_2 = a/b
print("%5d | %1.8f | %1.8f" %(i,f_1,f_2))
print("The value to 8 significant digits is %1.7f" %f_1)



    i | While loop | Explicit Formula
    9 | 2.00200133 | 2.00200133
The value to 8 significant digits is 2.0020013


We compared the value of f calculated by the while loop to the value of f calculated by the explicit formula both at $x=0.001$ to the 8 significant digits which is 2.0020013 (using %1.7f in the last line).

In [7]:
# Evaluate using for loop.
import math
# Initialize n.
n = 1
# Define x.
x = 0.001
# Evaluate f
f = 0
# Print header for the output table.
print("%+5s | %-10s | %-15s" %("i", "For loop", "Explicit Formula"))
print("=========================================================")
for i in range(1,7):
  f += ((2**i)/math.factorial(i))*x**(i-1)
# Evaluate f explicitely.
x = 0.001
a = math.e**(2*x) - 1
b = x
f_2 = a/b
print("%5d | %1.8f | %1.8f" %(i,f,f_2))
print("The value to 8 significant digits is %1.7f" %f)


    i | For loop   | Explicit Formula
    6 | 2.00200133 | 2.00200133
The value to 8 significant digits is 2.0020013


We campared the value of f calculated by the for loop to the value of f calculated by the explicit formula both at $x=0.001$ to the 8 significant digits which is 2.0020013 (using %1.7f in the last line).

# Problem 3

## Part a

Write two Python scripts, one using while loops and one using for loops, for calculating the sum
$$
        \sum^n_{k=1} k^2.
$$
Compare the results of each of your calculations with the formula
$$
    \sum^n_{k=1} k^2 = \frac{n(n+1)(2n + 1)}{6}
$$
using $n = 10$.

In [6]:
# Evaluating the sum using while loop.
# Initialize n.
n = 1
sum_1 = 0
# Print header for the output table.
print("%+5s | %-s | %-s" %("n", "While loop", "Explicit Formula"))
print("=========================================================")

# Start loop.
while (n <= 10):
  # Calculate sum_1
  sum_1 += n**2
  # Calculate the sum using explicit formula
  # Incerement n
  n += 1
# Calculate the sum using explicit formula
n=10  
sum_3 = (n*(n + 1)*(2*n+1))/6
print("%5d | %-10d | %-10d" %(n,sum_1, sum_3))

print("The value obtained by the while loop is %d" %sum_1)
print("The value obtained by the explicit formula is %d" %sum_3)

    n | While loop | Explicit Formula
   10 | 385        | 385       
The value obtained by the while loop is 385
The value obtained by the explicit formula is 385


In [5]:
# Evaluating the sum using for loop.
sum_2 = 0
# Print header for the output table.
print("%+5s | %s | %s" %("n", "For loop", "Explicit Formula"))
print("=========================================================")
for i in range(1,11):
  sum_2 += i**2

# Calculate the sum using explicit formula
n=10  
sum_3 = (n*(n + 1)*(2*n+1))/6
print("%5d | %-8d | %-10d" %(n,sum_2, sum_3))

print("The value obtained by the for loop is %d" %sum_2)
print("The value obtained by the explicit formula is %d" %sum_3)



    n | For loop | Explicit Formula
   10 | 385      | 385       
The value obtained by the for loop is 385
The value obtained by the explicit formula is 385


## Part b

Repeat for the sum
$$
        \sum_{k=1}^n \frac{1}{3^k}.
$$
Compare the values calculated by your two loops with value given by the formula
$$
        \sum_{k=1}^n \frac{1}{3^k} = \frac{1}{2}\left(1- \frac{1}{3^n} \right)
$$
for $n = 8$.

In [4]:
# Evaluating formula using while loop.
n = 1
sum = 0
# Print header for the output table.
print("%+5s | %-18s | %-21s" %("n", "While loop", "Explicit Formula"))
print("=========================================================")
while (n <= 8):
  sum += 1/3**n
  # incerement n.
  n += 1
# Evaluate the sum using the explicit formula.
n = 8
sum_3 = 0.5*(1 - (1/3**n))
print("%5d | %1.16f | %1.16f" %(n,sum, sum_3))
print("The value obtained by the while loop is %1.16f" %sum)
print("The value obtained by the explicit formula is %1.16f" %sum_3)


    n | While loop         | Explicit Formula     
    8 | 0.4999237921048620 | 0.4999237921048620
The value obtained by the while loop is 0.4999237921048620
The value obtained by the explicit formula is 0.4999237921048620


In [3]:
# Evaluating formula using for loop.
sum = 0
# Print header for the output table.
print("%+5s | %-21s | %-21s" %("n", "For loop", "Explicit Formula"))
print("=========================================================")
for i in range(1,9):
  sum += 1/3**i
# Evaluate the sum using the explicit formula.
n = 8
sum_3 = 0.5*(1 - (1/3**n))
print("%5d | %1.16f | %1.16f" %(n,sum, sum_3))
print("The value obtained by the for loop is %1.16f" %sum)
print("The value obtained by the explicit formula is %1.16f" %sum_3)


    n | For loop              | Explicit Formula     
    8 | 0.4999237921048620 | 0.4999237921048620
The value obtained by the for loop is 0.4999237921048620
The value obtained by the explicit formula is 0.4999237921048620
